examples/seismic/tutorials/08_DRP_schemes.ipynb

```diff
@@ -0,0 +1,653 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Custom finite difference coefficients in Devito\n",
+    "\n",
+    "## Introduction\n",
+    "\n",
+    "When taking the numerical derivative of a function in Devito, the default behaviour is for 'standard' finite difference weights (obtained via a Taylor series expansion about the point of differntation) to be applied. Consider the following example for some field $u(\\mathbf{x},t)$, where $\\mathbf{x}=(x,y)$. Let us define a computational domain/grid and differentiate our field with respect to $x$."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "from devito import Grid, TimeFunction\n",
+    "\n",
+    "# Create our grid (computational domain)\n",
+    "Lx = 10\n",
+    "Ly = Lx\n",
+    "Nx = 11\n",
+    "Ny = Nx\n",
+    "dx = Lx/(Nx-1)\n",
+    "dy = dx\n",
+    "grid = Grid(shape=(Nx,Ny), extent=(Lx,Ly))\n",
+    "\n",
+    "# Define u(x,y,t) on this grid\n",
+    "u = TimeFunction(name='u', grid=grid, time_order=2, space_order=2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, lets look at the output of $\\partial u/\\partial x$:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "-0.5*u(t, x - h_x, y)/h_x + 0.5*u(t, x + h_x, y)/h_x\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(u.dx)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "By default the 'standard' Taylor series expansion result, where `h_x` represents the $x$-direction grid spacing, is returned. However, there may be instances when a user wishes to use 'non-standard' weights when, for example, implimenting a dispersion-relation-preserving (DRP) scheme. See e.g. \n",
+    "\n",
+    "[1] Christopher K.W. Tam, Jay C. Webb (1993). ”Dispersion-Relation-Preserving Finite Difference Schemes for Computational Acoustics.” **J. Comput. Phys.**, 107(2), 262--281. https://doi.org/10.1006/jcph.1993.1142\n",
```

Comment:
I think this would be better but was just 'copying' what had been done in other notebooks for consistency. It would be nice to be able to use latex style referencing and bibliographies but such support for jupyter notebooks seems to require an additional package. Anyway, I'm open to suggestions.